lib/luacrypto_skcipher.c

```diff
@@ -0,0 +1,277 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+/***
+* Low-level Lua interface to the Linux Kernel Crypto API for symmetric
+* block ciphers (SKCIPHER).
+*
+* This module provides a `new` function to create SKCIPHER transform objects,
+* which can then be used for encryption and decryption with various block cipher
+* algorithms and modes.
+*
+* @module crypto_skcipher
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/crypto.h>
+#include <crypto/skcipher.h>
+#include <linux/err.h>
+#include <linux/scatterlist.h>
+#include <linux/slab.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+#include <lunatik.h>
+
+static const lunatik_class_t luacrypto_skcipher_class;
+
+typedef struct {
+	struct crypto_skcipher *tfm;
+	struct skcipher_request *req;
+
+	u8 *iv_data;
+	size_t iv_len;
+
+	u8 *work_buffer;
+	size_t work_buffer_len;
+
+	struct scatterlist sg_work;
+} luacrypto_skcipher_priv;
```

Comment:
```suggestion
} luacrypto_skcipher_t;
```
let's keep our style ;-)